lib/exercises.rb

```diff
@@ -2,19 +2,61 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
-end
-
-# This method will return the k most common elements
-# in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
-def top_k_frequent_elements(list, k)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  if strings.length == 0 #empty hash if no char in string
+    return []
+  end
+  
+  anagrams_hash = {} #where its a-going to get put
+  
+  anagram_array = []
+  
+  strings.each do |word| #take each word...
+    letters = word.split("").sort.join("") #...split into letters, and rejoin them
+    if anagrams_hash[letters].nil? #if string isnt in hash
+      anagrams_hash[letters] = word
```

Comment:
In this case this should be putting an array with just word in it.

```suggestion
      anagrams_hash[letters] = [word]
```